src/components/TagPicker.tsx

```diff
@@ -0,0 +1,133 @@
+import {useMemo, useState} from 'react';
+import type {OnyxCollection} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import useLocalize from '@hooks/useLocalize';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import * as PolicyUtils from '@libs/PolicyUtils';
+import type SafeAreaInsetProps from '@pages/SafeAreaInsetProps';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {PolicyTags, RecentlyUsedTags} from '@src/types/onyx';
+import OptionsSelector from './OptionsSelector';
+
+type TagPickerOnyxProps = {
+    /** Collection of tags attached to a policy */
+    policyTags: OnyxCollection<PolicyTags>;
```

Comment:
Since key takes only one item from collection
```suggestion
    policyTags: OnyxEntry<PolicyTags>;
```